satpy/composites/config_loader.py

```diff
@@ -170,120 +175,149 @@ def parse_config(self, configured_modifiers, composite_configs):
-            if sensor_name not in self.compositors:
-                self.load_sensor_composites(sensor_name)
-            if sensor_name in self.compositors:
-                comps[sensor_name] = DatasetDict(
-                    self.compositors[sensor_name].copy())
-                mods[sensor_name] = self.modifiers[sensor_name].copy()
-        return comps, mods
-
-    def _get_sensor_id_keys(self, conf, sensor_id, sensor_deps):
-        try:
-            id_keys = conf['composite_identification_keys']
-        except KeyError:
-            try:
-                id_keys = self._sensor_dataid_keys[sensor_deps[-1]]
-            except IndexError:
-                id_keys = minimal_default_keys_config
-        self._sensor_dataid_keys[sensor_id] = id_keys
-        return id_keys
-
-    def _load_config(self, composite_configs):
-        if not isinstance(composite_configs, (list, tuple)):
-            composite_configs = [composite_configs]
-
-        conf = {}
-        for composite_config in composite_configs:
-            with open(composite_config, 'r', encoding='utf-8') as conf_file:
-                conf = recursive_dict_update(conf, yaml.load(conf_file, Loader=UnsafeLoader))
-        try:
-            sensor_name = conf['sensor_name']
-        except KeyError:
-            logger.debug('No "sensor_name" tag found in %s, skipping.',
-                         composite_configs)
-            return
+def _load_config(composite_configs):
+    if not isinstance(composite_configs, (list, tuple)):
+        composite_configs = [composite_configs]
+
+    conf = {}
+    for composite_config in composite_configs:
+        with open(composite_config, 'r', encoding='utf-8') as conf_file:
+            conf = recursive_dict_update(conf, yaml.load(conf_file, Loader=UnsafeLoader))
+    try:
+        sensor_name = conf['sensor_name']
+    except KeyError:
+        logger.debug('No "sensor_name" tag found in %s, skipping.',
+                     composite_configs)
+        return
+
+    sensor_compositors = {}  # DatasetDict()
+    sensor_modifiers = {}
+
+    dep_id_keys = None
+    sensor_deps = sensor_name.split('/')[:-1]
+    if sensor_deps:
+        # get dependent
+        for sensor_dep in sensor_deps:
+            dep_comps, dep_mods, dep_id_keys = load_compositor_configs_for_sensor(sensor_dep)
+        # the last parent should include all of its parents so only add the last one
+        sensor_compositors.update(dep_comps)
+        sensor_modifiers.update(dep_mods)
+
+    id_keys = _get_sensor_id_keys(conf, dep_id_keys)
+    mod_config_helper = _ModifierConfigHelper(sensor_modifiers, id_keys)
+    configured_modifiers = conf.get('modifiers', {})
+    mod_config_helper.parse_config(configured_modifiers, composite_configs)
+
+    comp_config_helper = _CompositeConfigHelper(sensor_compositors, id_keys)
+    configured_composites = conf.get('composites', {})
+    comp_config_helper.parse_config(configured_composites, composite_configs)
```

Comment:
These two blocks look very similar to each other... could `parse_config` to the extraction?

Reply:
FYI this bit of code is unchanged. I'm fine refactoring more, but I'm not sure what you mean. These are two helper classes that are parsing two separate "things" (compositors versus modifiers).

Reply:
I was just thinking that the `conf.get('...', {})` could be done in the `parse_config` method

Reply:
Yeah it felt weird to pass the `conf` dict into the helper rather than just pass the dictionary I want to work with. This could probably be improved, but I think I'd rather say that for the future rather than have it get lost in this PR.